discord/embeds.py

```diff
@@ -383,15 +386,20 @@ def set_author(self, *, name, url=EmptyEmbed, icon_url=EmptyEmbed):
             The URL of the author icon. Only HTTP(S) is supported.
         """
 
-        self._author = {
-            'name': str(name)
-        }
+        if name is EmptyEmbed:
+            if url is not EmptyEmbed or icon_url is not EmptyEmbed:
+                raise ValueError("can't set author without name")
```

Comment:
This restriction makes no sense. The API does the validation in this case and a name is not actually required.

Reply:
Should I then only clear if all of those 3 arguments are set to EmptyEmbed?
I only did the restriction cause API silently ignores author if it doesn't have a name, but it does make sense to leave handling of that to the API so I'll change that.

Reply:
I made it clear author only when all 3 params are `EmptyEmbed`, let me know if you wanted it to clear whenever just `name` is `EmptyEmbed`.

Also, do you want me to make the default of `name` be `EmptyEmbed` too? IMO, it would make sense but I noticed that's not what was done for `set_thumbnail()` and `set_image()` so I wanted to ask first.

Reply:
Honestly it feels like if this were to be the case then the ergonomics of `clear_author()` are inherently better. Passing `EmptyEmbed` should only be if you want to clear specific attributes.